File: tests/machine/x/scala/cross_join_triple.scala
object cross_join_triple {
  case class Combo(n: Int, l: String, b: Boolean)

  val nums = List(1, 2)
  val letters = List("A", "B")
  val bools = List(true, false)
  val combos = for { n <- nums; l <- letters; b <- bools } yield Combo(n = n, l = l, b = b)
  def main(args: Array[String]): Unit = {
    println("--- Cross Join of three lists ---")
    for(c <- combos) {
      println(c.n + " " + c.l + " " + c.b)
    }
  }
}

File: tests/machine/x/cs/cross_join_triple.cs
using System;
using System.Collections.Generic;
using System.Linq;

class Program {
    static void Main() {
        List<int> nums = new List<int> { 1, 2 };
        List<string> letters = new List<string> { "A", "B" };
        List<bool> bools = new List<bool> { true, false };
        List<Combo> combos = (
    from n in nums
    from l in letters
    from b in bools
    select new Combo { n = n, l = l, b = b }
).ToList();
        Console.WriteLine("--- Cross Join of three lists ---");
        foreach (var c in combos) {
            Console.WriteLine(string.Join(" ", new [] { Convert.ToString(c.n), Convert.ToString(c.l), Convert.ToString(c.b) }));
        }
    }
    public class Combo {
        public int n;
        public string l;
        public bool b;
    }
    
    
    
}
